src/t_stream.c

```diff
@@ -2287,6 +2287,13 @@ void xreadCommand(client *c) {
                 ids[id_idx].ms = 0;
                 ids[id_idx].seq = 0;
             }
+            /* We change the '$' to the current last ID for this stream. this is
+             * Since later on when we unblock on arriving data - we would like to
+             * re-process the command and in case '$' stays we will spin-block forever.
+             */
+            robj *argv_streamid = createObjectFromStreamID(&ids[id_idx]);
+            decrRefCount(c->argv[i]);
+            c->argv[i] = argv_streamid;
```

Comment:
i think i'd rather do that just before deciding to block (calling blockForKeys), and not in the loop that processes arguments (we don't yet know what the command is gonna do).
both in terms of efficiency (generating that string is time consuming), and also due to possible sideeffects.
p.s. i think we have to use `rewriteClientCommandArgument`